src/main/java/org/truffleruby/interop/OutgoingForeignCallNode.java

```diff
@@ -152,7 +152,7 @@ protected int getCacheLimit() {
         return getContext().getOptions().INTEROP_EXECUTE_CACHE;
     }
 
-    protected abstract class OutgoingNode extends Node {
+    protected abstract static class OutgoingNode extends Node {
```

Comment:
Should the classes below also be static @chrisseaton?
Maybe OutgoingNode should extend `RubyBaseNode` and get passed the `name` explicitly?

Reply:
They should be static if they can be. I guess @nirvdrum used a tool to figure this out though?

Reply:
They need a few more changes as I mentioned in my comment, so a tool wouldn't detect them as "can be declared static trivially".

Reply:
Correct. I customized the "intentions" in IntelliJ and ran them. Where changes could be trivially applied, I did so. Where more work was needed to "fix" a class of problem, I went through those. But I didn't actively look for other candidates for any suggested fix.